### PR TITLE
Gamemode code cleanup.

### PR DESCRIPTION
**Pull request by DominikPanic**
Cleans up the gamemode and a bit of ticker code.

apparently can't reopen after rebasing sorry

### CODE REVIEW

File: code/__DEFINES/mode.dm [hunk 1]
```diff
@@ -51,12 +46,61 @@
...
+
+#define SHUTTLE_TIME_LOCK 		10 MINUTES
+#define SHUTTLE_LOCK_COOLDOWN 	10 MINUTES
+#define SHUTTLE_LOCK_TIME_LOCK 	45 MINUTES
+
+#define COOLDOWN_COMM_REQUEST 	5 MINUTES
+#define COOLDOWN_COMM_MESSAGE 	1 MINUTES
+#define COOLDOWN_COMM_CENTRAL 	30 SECONDS
+
+#define XENO_AFK_TIMER			300
```
Comment (Rohesie) [inline code/__DEFINES/mode.dm]:
```suggestion
#define XENO_AFK_TIMER			300 // Number of life proc ticks. In for re-doing.
```

File: code/game/gamemodes/distress.dm [hunk 2]
```diff
@@ -0,0 +1,702 @@
...
+	required_players = 2
+	flags_round_type = MODE_INFESTATION|MODE_FOG_ACTIVATED
+	flags_landmarks = MODE_LANDMARK_SPAWN_XENO_TUNNELS|MODE_LANDMARK_SPAWN_MAP_ITEM
+
+	var/list/survivors = list()
+
+	var/xeno_required_num 	  = 1
+	var/xeno_starting_num 	  = 0
+	var/surv_starting_num 	  = 0
+	var/merc_starting_num 	  = 0
```
Comment (Rohesie) [inline code/game/gamemodes/distress.dm]: Remove the legacy var?

File: code/game/gamemodes/distress.dm [hunk 3]
```diff
@@ -0,0 +1,702 @@
...
+						/obj/item/ammo_magazine/shotgun/buckshot = round(scale * 10),
+						/obj/item/ammo_magazine/shotgun/flechette = round(scale * 10),
+						/obj/item/cell/lasgun/M43 = round(scale * 25),
+
+						/obj/item/weapon/combat_knife = round(scale * 30),
+						/obj/item/weapon/throwing_knife = round(scale * 10),
+						/obj/item/storage/box/m94 = round(scale * 10),
+
+						/obj/item/attachable/flashlight = round(scale * 25),
+						/obj/item/attachable/bayonet = round(scale * 25),
```
Comment (Rohesie) [inline code/game/gamemodes/distress.dm]:
```suggestion
						/obj/item/attachable/bayonet = round(scale * 25)
```

File: code/game/gamemodes/distress.dm [hunk 4]
```diff
@@ -0,0 +1,702 @@
...
+						/obj/item/explosive/grenade/cloakbomb = round(scale * 3),
+						/obj/item/storage/box/m94 = round(scale * 30),
+						/obj/item/device/flashlight/combat = round(scale * 5),
+						/obj/item/clothing/mask/gas = round(scale * 10)
+						)
+
+		CG.contraband = list(
+						/obj/item/weapon/gun/smg/ppsh = round(scale * 4),
+						/obj/item/weapon/gun/shotgun/double = round(scale * 2),
+						/obj/item/weapon/gun/shotgun/pump/bolt = round(scale * 2),
```
Comment (Rohesie) [inline code/game/gamemodes/distress.dm]:
```suggestion
						/obj/item/weapon/gun/shotgun/pump/bolt = round(scale * 2)
```

File: code/controllers/stonedmc/subsystem/ticker.dm [hunk 5]
```diff
@@ -55,16 +55,19 @@ SUBSYSTEM_DEF(ticker)
 	switch(current_state)
 		if(GAME_STATE_STARTUP)
 			if(Master.initializations_finished_with_no_players_logged_in)
+				start_at = world.time + ((CONFIG_GET(number/lobby_countdown) * 10) * 2)
+			else
 				start_at = world.time + (CONFIG_GET(number/lobby_countdown) * 10)
```
Comment (Menshin) [inline code/controllers/stonedmc/subsystem/ticker.dm]: Keeping this line and removing the three lines before, will set the counter when the first client connect to the server.
Also, if you do that, you can remove the `start_at = world.time + (CONFIG_GET(number/lobby_countdown) * 10)` set during `initialize`.

File: code/controllers/stonedmc/subsystem/ticker.dm [hunk 6]
```diff
@@ -81,10 +84,10 @@ SUBSYSTEM_DEF(ticker)
 
 		if(GAME_STATE_SETTING_UP)
 			if(!setup())
-				//setup failed
 				current_state = GAME_STATE_STARTUP
-				start_at = world.time + (CONFIG_GET(number/lobby_countdown) * 10)
 				time_left = null
+				start_at = world.time + (CONFIG_GET(number/lobby_countdown) * 10)
```
Comment (Menshin) [inline code/controllers/stonedmc/subsystem/ticker.dm]: You can also remove this one if you choose to make the suggestion above.

File: code/game/gamemodes/distress.dm [hunk 7]
```diff
@@ -0,0 +1,695 @@
...
+	scale_gear()
+
+	addtimer(CALLBACK(SSticker.mode, .proc/map_announce), 5 SECONDS)
+
+
+
+/datum/game_mode/distress/proc/map_announce()
+	switch(SSmapping.config.map_name)
+		if(MAP_LV_624)
+			command_announcement.Announce("A faint distress signal has been picked up by our scanners, which have tracked the source to a third generation colony, known as LV-624. Through use of bluespace drive tech, the [CONFIG_GET(string/ship_name)] has jumped within range of the colony. TGMC, gear up and get ready to respond!", "[CONFIG_GET(string/ship_name)]")
```
Comment (spookydonut) [inline code/game/gamemodes/distress.dm]: this should probably be moved into the map configs.

File: code/game/gamemodes/distress.dm [hunk 8]
```diff
@@ -0,0 +1,695 @@
...
+		if(MODE_INFESTATION_M_MAJOR)
+			musical_track = pick('sound/theme/winning_triumph1.ogg','sound/theme/winning_triumph2.ogg')
+		if(MODE_INFESTATION_X_MINOR)
+			musical_track = pick('sound/theme/neutral_melancholy1.ogg','sound/theme/neutral_melancholy2.ogg')
+		if(MODE_INFESTATION_M_MINOR)
+			musical_track = pick('sound/theme/neutral_hopeful1.ogg','sound/theme/neutral_hopeful2.ogg')
+		if(MODE_INFESTATION_DRAW_DEATH)
+			musical_track = pick('sound/theme/nuclear_detonation1.ogg','sound/theme/nuclear_detonation2.ogg') //This one is unlikely to play.
+
+	to_chat(world, musical_track)
```
Comment (spookydonut) [inline code/game/gamemodes/distress.dm]: send_sound

File: code/game/gamemodes/distress.dm (hunk 7 above)
Comment (DominikPanic) [reply code/game/gamemodes/distress.dm]: It's gamemode specific though

Comment (spookydonut) [reply code/game/gamemodes/distress.dm]: It's map specific like you could store this text in the map json and replace the switch with just a reference to map config. Ideally we should be able to just add maps via config without having to change code like this.

**Pull request merged**